Tic Tac Toe.py

```diff
@@ -0,0 +1,175 @@
+    print('''===========================
+Welcome to Tic Tac Toe
+GAME RULES:
+Each player can place one mark (or stone) per turn on the 3x3 grid
+The WINNER is who succeeds in placing three of their marks in a
+* horizontal,
+* vertical or
+* diagonal row
+Let's start the game''')
+    print('------')
+    print(' | | ')
+    print('------')
+    print(' | | ')
+    print('------')
+    print(' | | ')
+    print('------')
+    print('='*27)
+
+def first_player():
+       lst = [0, 1]
+       random.shuffle(lst)
+       return lst
+
+
+def choose_sign(player1, player2):
+    while True:
+        if player1 == 1:
+            player = 'Player1'
+            choices = ['x', 'o']
+            sypher = input('Random choose {} as first player. {} please insert your sign: x or o \n'.format(player, player))
+            if sypher != choices[0] and sypher != choices[1]:
+                print('You have to insert x or o')
+                continue
+            else:
+                if sypher == 'x':
+                    return ['x', 'o']
+                else:
+                    return ['o', 'x']
+
+        elif player2 == 1:
+            player = 'Player2'
+            choices = ['x', 'o']
+            sypher = input('Random choose {} as first player. {} please insert your sign: x or o \n'.format(player, player))
+            if sypher != choices[0] and sypher != choices[1]:
+                print('You have to insert x or o ')
+                continue
+            else:
+                if sypher == 'x':
+                    return ['o', 'x']
+                else:
+                    return ['x', 'o']
+
+
+def print_progress(dic):
+    print('------')
+    print('{}|{}|{}'.format(dic['1'], dic['2'], dic['3']))
+    print('------')
+    print('{}|{}|{}'.format(dic['4'], dic['5'], dic['6']))
+    print('------')
+    print('{}|{}|{}'.format(dic['7'], dic['8'], dic['9']))
+    print('------')
+
+
+def win(dic, sypher):
+
+    if dic['1'] == dic['2'] == dic['3'] == sypher:
+        return True
+    if dic['4'] == dic['5'] == dic['6'] == sypher:
+        return True
+    if dic['7'] == dic['8'] == dic['9'] == sypher:
+        return True
+
+    if dic['1'] == dic['4'] == dic['7'] == sypher:
+        return True
+    if dic['2'] == dic['5'] == dic['8'] == sypher:
+        return True
+    if dic['3'] == dic['6'] == dic['9'] == sypher:
+        return True
+
+    if dic['1'] == dic['5'] == dic['9'] == sypher:
+        return True
+    if dic['3'] == dic['5'] == dic['7'] == sypher:
+        return True
+
+    return False
+
+def exit_requests(player1, player2):
+    for i in dic:
+        dic[i] = ' '
+    if player1 == 1:
+        print('Player2 `s turn')
+        return (0, 1)
+    elif player2 == 1:
+        print('Player1 `s turn')
+        return (1, 0)
+
+def main():
+    while True:
+        next_game = True
+        while next_game:
+            next_game = False
+            print_template()
+            player1, player2 = first_player()
```

Comment:
I suggest to not use pairs of variables that are connected like this. You can easily express this with one variable `player` which will have either 1 or 0 and next player will be just `player = (player + 1) % 2`.

Reply:
Why is this connections wrong ?

Reply:
What is difference between return (1, 0) vs return player1, player2; 
where player1 = 1 and player2 = 0

Reply:
It is always better to have fewer variables than more. Less code = fewer bugs. When you duplicate some state in your application (even inverted like here) it may happen that there will be a bug and you will have suddenly both players `0`. When you have only one variable this cannot happen.